src/js/letters/actions/letters.js

```diff
@@ -76,35 +75,37 @@ export function getAddressFailure() {
 
 export function getMailingAddress() {
   return (dispatch) => {
-    apiRequest(
+    return apiRequest(
       '/v0/address',
       null,
-      response => {
+      // on fetch success
+      (response) => {
         const responseCopy = Object.assign({}, response);
-        const address = Object.assign({}, response.data.attributes.address);
-        // Translate military-only fields into generic ones; we'll translate them back later if necessary
-        if (address.type === ADDRESS_TYPES.military) {
-          address.city = address.militaryPostOfficeTypeCode;
-          address.stateCode = address.militaryStateCode;
-          address.countryName = 'USA';
-          delete address.militaryPostOfficeTypeCode;
-          delete address.militaryStateCode;
+        const addressCopy = Object.assign({}, response.data.attributes.address);
+        // Translate military-only fields into generic ones; we'll translate
+        // them back later if necessary
+        if (addressCopy.type === ADDRESS_TYPES.military) {
+          addressCopy.city = addressCopy.militaryPostOfficeTypeCode;
+          addressCopy.stateCode = addressCopy.militaryStateCode;
+          addressCopy.countryName = 'USA';
+          delete addressCopy.militaryPostOfficeTypeCode;
+          delete addressCopy.militaryStateCode;
         }
-        responseCopy.data.attributes.address = address;
-
-        dispatch({
+        responseCopy.data.attributes.address = addressCopy;
```

Comment:
Eh, because  `responseCopy` is only a shallow copy (with `Object.assign()`), this technically isn't necessary, but it does read a lot nicer with it.